src/bridge.h

```diff
@@ -52,6 +75,8 @@ struct BridgeSpec {
 	StringID transport_name[2];         ///< description of the bridge, when built for road or rail
 	std::vector<std::vector<PalSpriteID>> sprite_table; ///< table of sprites for drawing the bridge
 	uint8_t flags;                         ///< bit 0 set: disable drawing of far pillars.
+	uint8_t ctrl_flags;          ///< control flags
+	uint8_t pillar_flags[12];    ///< bridge pillar flags: 6 x pairs of x and y flags
```

Comment:
These should then become `BridgeControlFlags` and `BridgePiecePillarFlags` instead of `uint8_t`.

I'm not keen on having 6 pairs of values as an array of 12. Consider having two arrays, one for x and one for y, or putting both x and y flags into a struct and using an array of 6.

Also, prefer `std::array<>` when adding new arrays to the code.